### ISSUE TITLE
BUG - sev-2 - All - Keyboard navigation on subtasks, large panels, cancel popups often goes to hidden layers, gets stuck

### ISSUE THREAD

**Issue by TKDickson** [department-of-veterans-affairs/va-mobile-app#9476]
## What happened?

Keyboard navigation on both iOS and Android will navigate 'behind' the panel on subtasks, large panels, and cancel popups. Often on iOS you can still get to the desired content/actions; on Android it's more difficult (and sometimes not possible, depending on the screen). 

Here's [updating contact info on Android, with keyboard.](https://drive.google.com/file/d/1jMyG_GDmFjQhr2DAWfuC-czzlrlsIEEi/view?usp=drive_link) The times when I'm 'stuck' for a long while and then magically get unstuck, it's because I've used my finger to tap directly onto the screen (obviously not a viable workaround for users).

## Specs:

- Device:
- OS:
- User Account:
- Accessibility State: Turn on Keyboard full access for your device

## Steps to Reproduce

(See video in description, above)
* Turn on keyboard full access for your device (you can log into the mobile app first if you like)
* Start any workflow - editing and saving contact info is a good proxy - in which you're going to have large panels, fullscreen subtasks, or native popups -- layers on the UI where typically there's a visible "thing behind" and "thing in front"
* Attempt to do that workflow with a keyboard. Some of them are possible to 'get into' the UI in front with things like "shift+tab 20 times to get onto the popup"; others are not possible to get the keyboard onto at all.

## Desired behavior

Navigation/Focus should be on the 'top' layer of the UI (large panel, native cancel popup, etc), not allowed to go behind it, or even worse - trapped behind it.

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: High 
- Frequency: Low 

- 2 - High

## Screen shot(s) and additional information

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  

</details>

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [ ] Labels added (front-end, back-end, global, Health, relevant feature, accessibility, etc)
- [ ] Estimate of 1 added (for front-end tickets)
- [ ] Added either to the relevant feature epic (if found during new feature implementation), or the relevant team's bug epic ([Global](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8302), [Health & Benefits](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8303), [API](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8304), [QART](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8305))

**Comment by alexandec** [comment]
I tested this with iOS 18 and I'm still seeing the issue. I can reliably use the tab key to place focus on the Cancel button, but using the arrow keys frequently navigates behind the large panel in front.

**Comment by alexandec** [comment]
Some background info on keyboard navigation:
- It's possible for VoiceOver/TalkBack focus to work fine, but for keyboard navigation the focus may not work correctly on the same elements due to a difference in implementation between screen reader and keyboard navigation
- We can't detect whether the user has FKA (Full Keyboard Access) enabled on iOS so we can't enable a workaround only for FKA
- Any keyboard navigation workarounds must be carefully tested with screen readers to ensure they don't introduce regressions
- See the comments on [#9474](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/9474) for additional investigation and info on keyboard navigation

**Comment by rbontrager** [comment]
@theodur Your solution caused another issue.  Basically while you can now navigate to the street address line 1 in android in your build, you still can't get into and edit the field with the keyboard.

**Comment by theodur** [comment]
@rbontrager When I was looking into the issue a while ago, it seems like this is a known issue with React Native and Android, where physical keyboard detection isn't picked up on Android.

https://reactnative.dev/docs/textinput#onkeypress
> Note: on Android only the inputs from soft keyboard are handled, not the hardware keyboard inputs.

https://github.com/facebook/react-native/issues/30464

This will probably need its own ticket to go in depth with the Android text input issue. For now, the changes in my PR at least prevent users from getting stuck on pop-ups, large panels, etc. and allows them to navigate through the whole form which is an improvement from the current experience.

**Comment by timwright12** [comment]
Opening this back up since the PR isn't closed out yet

**Comment by digitalken** [comment]
@dumathane - Is there a specific build for this to test? I don't see anything in my list of builds on testflight or firebase that seems to refer to keyboard nav or this bug number.

**Comment by timwright12** [comment]
@digitalken set up a build for you: https://github.com/department-of-veterans-affairs/va-mobile-app/actions/runs/14089176326

**Comment by digitalken** [comment]
@dumathane - I ran into issues with both iOS and Android (just tested Samsung in Android) where I'm still able to be navigating around on the screen behind the slide up screen.  I made a couple of recordings, one for iOS and one for Android.  Interestingly with iOS, when I was on a slide up screen and it had a phone call link, when I brought the link up I wasn't able to navigate to it and dial it up, unless I'm still too green with the keyboard.  How should I handle the recordings?  They are 4 and 6 MB files.

**Comment by digitalken** [comment]
@dumathane - I have a couple of videos of what I'm seeing.  The Android video has sound, so go ahead and turn up your sound to hear it. I narrate what I'm doing.  The iOS video has no sound.  For both, I noticed that I'm still navigating with arrows on the background screen but I can use the tab key to jump into the popup.  For both, you can see some activity going on in the background screen as I tap arrow keys on the keyboard.  Now for iOS, and I'm not sure this is within the scope of this bug, but once I get the native phone number dialer to pop up, there is no way for me (that I found, anyway) to tap it with the keyboard to initiate the call.  I'm stuck at that point.  I can't seem to get out of that screen, either, using the keyboard. The first file below is the Android / Samsung file.

https://github.com/user-attachments/assets/8fa7681a-8b1d-4953-a1d8-75879cefc047

https://github.com/user-attachments/assets/b54aa678-c634-414a-9344-1300426ce4eb

**Comment by timwright12** [comment]
@dumathane If this requires more rework, up to you if you want to keep it going or just backlog the ticket

**Comment by timwright12** [comment]
Frontend estimate: 8pts

**Comment by b-rocha** [comment]
@dumathane @timwright12 Where did we land on this one? Does this make sense to take on or is this not worth the rework?

**Comment by timwright12** [comment]
I'd like to see us try and fix this one

**Comment by dftony** [comment]
Got a PR up that addresses largePanels. 

Currently investigating FullScreenSubtasks, it looks to be a bit more involved since there are issues around text inputs in addition to the hidden layer bug.

**Comment by dftony** [comment]
**Updates, summaries, and more findings**: Keyboard navigation fails on Android across large panels, fullScreenSubtasks, and multiStepSubtasks, while iOS works correctly.

The keyboard navigation issue for large panels has been [resolved with this PR](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/10919) ✅ 

I’ve implemented a [working fix](https://github.com/department-of-veterans-affairs/va-mobile-app/compare/bug/9476-fsSubtasks-inputs?diff=split&w) for fullScreenSubtasks (includes a bug fix with keyboard navigation textInput focus). There are about 14 instances that use this in the code. ✅ 

MultiStepSubtasks remain problematic and have been tricky due to their nested navigation structure. There are 3 instances (FileRequest, SubmitEvidence, SubmitMileageTravelPay)

It may be worthwhile to create a new bug ticket solely for multiStepSubtasks to prevent more bloat on this thread.

**Comment by dftony** [comment]
The update I originally considered feels a bit too invasive for a bug of this scope, and I suspect there's a cleaner solution that I haven't identified yet. I'm going to close this issue for now, but linking to a PR that addresses most of these concerns for reference.

https://github.com/department-of-veterans-affairs/va-mobile-app/pull/11228
